packages/trader/src/AppV2/Components/TradeParameters/Stake/stake-input.tsx

```diff
@@ -0,0 +1,330 @@
+
+    const [details, setDetails] = React.useState({
+        commission,
+        error_1: first_payout_error,
+        error_2: second_payout_error,
+        first_contract_payout,
+        is_first_payout_exceed: !!first_payout_error && first_contract_payout > max_payout,
+        is_second_payout_exceed: !!second_payout_error && second_contract_payout > max_payout,
+        second_contract_payout,
+        max_payout,
+        max_stake,
+        min_stake,
+        stop_out,
+    });
+
+    // Parallel proposal without subscription
+    // For Rise/Fall and all Digits we should do 2 proposal requests
+    const should_send_multiple_proposals = contract_types.length > 1 && !is_multiplier;
+    const has_both_errors = !!details.error_1 && !!details.error_2;
+    // Need for cases with Rise/Fall and Digits, when only one response contains error and we should allow to save the value
+    const should_show_stake_error =
+        !should_send_multiple_proposals || (should_send_multiple_proposals && has_both_errors);
+
+    const proposal_req_1 = getProposalRequestObject({
+        new_values: proposal_request_values,
+        trade_store,
+        trade_type: contract_types[0],
+    });
+    const { data: response_1, is_fetching: is_fetching_1 } = useDtraderQuery<Parameters<TOnProposalResponse>[0]>(
+        [
+            'proposal',
+            ...Object.entries(proposal_request_values).flat().join('-'),
+            `${proposal_request_values?.amount}`,
+            JSON.stringify(proposal_req_1),
+            contract_types.join('-'),
+        ],
+        proposal_req_1,
+        {
+            enabled: is_open,
+        }
+    );
+
+    const proposal_req_2 = getProposalRequestObject({
+        new_values: proposal_request_values,
+        trade_store,
+        trade_type: contract_types[1],
+    });
+    const { data: response_2, is_fetching: is_fetching_2 } = useDtraderQuery<Parameters<TOnProposalResponse>[0]>(
+        [
+            'proposal',
+            ...Object.entries(proposal_request_values).flat().join('-'),
+            `${proposal_request_values?.amount}`,
+            JSON.stringify(proposal_req_2),
+            contract_types.join('-'),
+        ],
+        proposal_req_2,
+        {
+            enabled: is_open && should_send_multiple_proposals,
+        }
+    );
+
+    const is_loading_proposal = is_fetching_1 || (should_send_multiple_proposals && is_fetching_2);
+
+    React.useEffect(() => {
+        const onProposalResponse: TOnProposalResponse = response => {
+            const { error, proposal } = response;
+
+            // In case if the value is empty we are showing custom error text from FE (in onSave function)
+            if (proposal_request_values.amount === '') {
+                setStakeError('');
+                return;
+            }
+
+            // Edge cases for Vanilla and Turbos
+            if (is_vanilla && error?.details?.barrier_choices) {
```

Comment:
Lines from 141 to 157 was taken from _trade-store_, where this cases were handled inside of `onProposalResponse`